posts/2020-10-23-beta-20.0.0.12.adoc

```diff
@@ -0,0 +1,452 @@
+
+We have two beta packages for link:{url-about}[Open Liberty]:
+
+* <<allbeta, All Beta Features>>: a larger package that contains all Open Liberty beta features (including Jakarta EE 9 beta features) and GA features and functions.
+* <<jakarta, Jakarta EE 9 Beta Features>>: a lightweight package that contains only the Jakarta EE 9 features.
+
+This means that you can now try out our in-development Open Liberty features by just adding the relevant coordinates to your build tools.
+
+If you try either package, <<feedback, let us know what you think>>.
+[#allbeta]
+== All Beta Features package
+
+The All Beta Features package includes the following beta features:
+
+* <<mp4, Support for MicroProfile 4.0>>
+* <<conf, Support for MicroProfile Config 2.0>>
+* <<jwt, Support for MicroProfile JWT 1.2>>
+* <<lra, Support for MicroProfile LRA>>
+* <<java15, Support for Java SE 15>>
+
+[#mp4]
+=== Support for MicroProfile 4.0
+
+The Open Liberty 20.0.0.12-beta provides most feature updates for MicroProfile 4.0-M1, this includes updates to MicroProfile Config, Fault Tolerance, Metrics, Health, Open Tracing, Open API and JWT, detailed below. While MicroProfile Rest Client is part of the MicroProfile 4.0-M1 specification, this feature narrowly missed the beta train and is scheduled for delivery in milestone 2.
+
+MicroProfile 4.0 brings the following feature updates to Open Liberty:
+
+* Config 2.0
+* Fault Tolerance 3.0
+* Metrics 3.0
+* Health 3.0
+* Open API 2.0
+* JWT 1.2
+* Open Tracing 2.0
+
+To enable the new beta features in your app, add them to your `pom.xml`:
+
+[source, xml]
+----
+    <dependencies>
+        <dependency>
+            <groupId>org.eclipse.microprofile</groupId>
+            <artifactId>microprofile</artifactId>
+            <version>4.0</version>
+            <type>pom</type>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+----
+
+For more information about the MicroProfile updates included in this milestone, visit the link:https://download.eclipse.org/microprofile/microprofile-4.0-M1/microprofile-spec-4.0-M1.html#microprofile4.0-M1[release page].
+
+[#conf]
+=== Support for MicroProfile Config 2.0
+
+link:https://github.com/eclipse/microprofile-config[MicroProfile Config] is a specification which Open Liberty implements to allow application configuration from different locations (ConfigSources). In the previous versions, Open Liberty has implemented the specification in its own codebase. However, for MicroProfile Config 2.0 it consumes link:https://github.com/smallrye/smallrye-config[SmallRye's implementation] of the specification.
+
+MicroProfile Config 2.0 is a major version release (from 1.4) which introduces API/SPI and functional changes.
```

Comment:
"introduces functional changes and API/SPI incompatible changes." (match the order of the section that follow)